core/chains/evm/logpoller/orm.go

```diff
@@ -467,12 +467,13 @@ func (o *ORM) SelectIndexedLogsCreatedAfter(address common.Address, eventSig com
 	// Add 1 since postgresql arrays are 1-indexed.
 	err := q.Select(&logs, `
 		SELECT * FROM evm_logs 
-			WHERE evm_logs.evm_chain_id = $1
-			AND address = $2 AND event_sig = $3
+			WHERE evm_chain_id = $1
+			AND address = $2 
+			AND event_sig = $3
 			AND topics[$4] = ANY($5)
-			AND created_at > $6
+			AND block_number >= (SELECT COALESCE(block_number, 0) FROM evm_log_poller_blocks WHERE evm_chain_id = $1 and block_timestamp > $6 ORDER BY block_number LIMIT 1)
 			AND (block_number + $7) <= (SELECT COALESCE(block_number, 0) FROM evm_log_poller_blocks WHERE evm_chain_id = $1 ORDER BY block_number DESC LIMIT 1)
-			ORDER BY created_at ASC`, utils.NewBig(o.chainID), address, eventSig.Bytes(), topicIndex+1, pq.ByteaArray(topicValuesBytes), after, confs)
+			ORDER BY block_number ASC`, utils.NewBig(o.chainID), address, eventSig.Bytes(), topicIndex+1, pq.ByteaArray(topicValuesBytes), after, confs)
```

Comment:
Understand it might be tricky, can we have a test case for it?

Reply:
@samsondav please take a look, wrote a couple of tests. Additionally, I reverted changes from `orm.go` and run those tests against the old implementation - they did not pass, because of `created_at` behavior - which is expected ;)